asserts/snap_asserts_test.go

```diff
@@ -154,6 +157,44 @@ func (sds *snapDeclSuite) TestDecodeInvalid(c *C) {
 
 }
 
+func (sds *snapDeclSuite) TestDecodePlugs(c *C) {
+	encoded := `type: snap-declaration
+authority-id: canonical
+series: 16
+snap-id: snap-id-1
+snap-name: first
+publisher-id: dev-id1
+plugs:
+  interface1:
+    allow-auto-connection:
+      slot-attributes:
+        a1: /foo/.*
+  interface2:
+    allow-connection:
+      slot-attributes:
+        a2: /foo/.*
```

Comment:
allow/deny-installation doesnt make a lot of sense at the snap-declaration level, I fear it would be a bit confusing, I can add the other kind of fields and subrules though